baselines/packages/mimir/test/parameter-properties/default/test.ts.lint

```diff
@@ -0,0 +1,83 @@
+export class Hello {
+    constructor(private world: string) {}
+                ~~~~~~~~~~~~~~~~~~~~~     [error parameter-properties: Parameter properties have been disallowed.]
+}
+
+export class Hello extends World {
+    constructor(private world: string) {
+                ~~~~~~~~~~~~~~~~~~~~~    [error parameter-properties: Parameter properties have been disallowed.]
+        super();
+    }
+}
+
+export class Welcome {
+    constructor(private home: string = 'mom!') {}
+                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~     [error parameter-properties: Parameter properties have been disallowed.]
+}
+
+export class Hello {
+    public hello: string;
+    constructor(hello: string = 'mom!') {
+        this.hello = hello;
+    }
+}
+
+export class Hello extends World {
+    public hello: string;
+    constructor(hello: string = 'mom!') {
+        super();
+        this.hello = hello;
+    }
+}
+
+/* Tests for multiple access modifiers */
+export class Hello extends World {
+    private readonly hello: string;
+    constructor(hello: string = 'mom!') {
+        super();
+        this.hello = hello;
+    }
+}
+
+export class Hello extends World {
+    constructor(private readonly hello: string = 'mom!') {
+                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~    [error parameter-properties: Parameter properties have been disallowed.]
+        super();
+    }
+}
+
+class Foo {
+    fizz: string;
+    constructor(fizz: string) {
+        this.fizz = fizz + ' buzz';
+    }
+}
+
+class Foo extends Bar {
+    fizz: string;
+    constructor(fizz: string) {
+        super();
+        this.fizz = fizz + ' buzz';
+    }
+}
+
+/* when-possible config should ignore this case because param is not the first thing to be assigned to the prop */
+class Foo {
+    fizz: string;
+    constructor(fizz: string) {
+        this.fizz = 'buzz';
+        this.fizz = fizz;
+    }
+}
+
+class Foo {
+    constructor() {}
+}
+
+class Foo {
+    private bar: string;
+    constructor(bar: string, private fizz: boolean) {
```

Comment:
please add a test with an optional parameter property to ensure the `?` is also added to the property declaration.